doc/proposals/openshift-4.3/operator-testing-tool.md

```diff
@@ -0,0 +1,240 @@
+
+
+Tasks for this story include:
+ * determine the correct validation library to use
+ * add support for a bundle directory flag to the CLI
+ * change the scorecard code to use that validation library when validating a CR, CSV, or given a bundle directory
+ * format the validation output to present to the CLI user in text or json
+
+Rough point estimate in this story is 5 points.
+
+#### Story 4 - Update scorecard Documentation
+Tasks:
+ * Document changes to CLI
+ * Document new output formats
+ * Document changes to configuration
+ * Document breaking changes and removals
+
+
+
+### Implementation Details/Notes/Constraints
+
+A scheme for applying the labels to the tests would need to be developed as well as label names to categorize tests.
+
+The “source of truth” for validation would need to be established such as which library the scorecard would use to perform CR/CSV/bundle validation.
+
+If no runtime tests are specified, then the scorecard would only run the static tests and not depend on a running cluster.
+
+### Risks and Mitigations
+
+The scorecard would implement a version flag in the CLI to allow users to migrate from current functionality to the proposed functionality (e.g. v1alpha2):
+ * operator-sdk scorecard --version v1alpha2
+
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+maturity levels.
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
```

Comment:
I think it is not required. 
- operator-sdk still in init dev. See that its tag is < 1.0.0
-In POV the biggest part of users has not been using it anyway 
- The CI in the community-operators repos will be affected as well but after a new release with the changes, it should be still working fine by using the previous version until we are able to update there. Maybe is required sinilize the responsible for this repo about our intention for they are able to follow up and plan the changes.

Reply:
While we're not yet to 1.0.0, the CVP has automation around scorecard that would break suddenly if we didn't have a deprecation and removal plan that was communicated in advance. In this case, I think it's important to document what we're doing:
1. Adding a new `--version` flag
2. Adding a new result output spec, `v1alpha2`, based on the changes described in this proposal.
3. Setting the default `--version` to `v1alpha1` to avoid breaking existing CLI users.
4. Deprecating `v1alpha1`, and informing users to move to `v1alpha2` by setting `--version=v1alpha2` on the command line.
5. In a subsequent release, removing support for `v1alpha1`.

Reply:
I'll update this section in a new version of the proposal.